ui/app/AppLayouts/Chat/controls/community/TokenPanel.qml

```diff
@@ -0,0 +1,88 @@
+import QtQuick 2.14
+import QtQuick.Layouts 1.14
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Components 0.1
+
+import shared.controls 1.0
+
+ColumnLayout {
+    id: root
+
+    property int mode: HoldingTypes.Mode.Add
+    property alias tokenName: item.name
+    property alias tokenShortName: item.shortName
+    property alias tokenImage: item.iconSource
+    property alias amountText: amountInput.text
+    property alias amount: amountInput.amount
+    property alias tokenCategoryText: tokenLabel.text
+    property alias addOrUpdateButtonEnabled: addOrUpdateButton.enabled
+    property alias allowDecimals: amountInput.allowDecimals
+    readonly property bool amountValid: amountInput.valid && amountInput.text.length > 0
+
+    signal addClicked
+    signal updateClicked
+    signal removeClicked
+
+    function setAmount(amount) {
+        amountInput.setAmount(amount)
+    }
+
+    QtObject {
+        id: d
+
+        // values from design
+        readonly property int buttonsHeight: 44
+        readonly property int defaultSpacing: 16
```

Comment:
I am just wondering - creating properties impacts performance, since in this case is only referenced once, can't we just set 16 directly and in line 63 refer to root.spacing instead?

Reply:
Removed `defaultSpacing` as suggested!